kitsune/ai/questions/utils.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import transaction
+
+from kitsune.flagit.models import FlaggedObject
+from kitsune.products.models import Topic
+from kitsune.questions.models import Question
+from kitsune.users.models import Profile
+
+
+def flag_as_spam(question: Question, reason: str) -> None:
+    """
+    Flags the given question as spam with the given reason.
+    """
+    if question.is_spam:
+        return
+
+    sumo_bot = Profile.get_sumo_bot()
+
+    with transaction.atomic():
+        # Mark the question as spam, and assign it the "Undefined" topic.
+        question.topic = Topic.active.get(title="Undefined")
```

Comment:
Although I am not against this atm, at the same time I am not sure whether we should this or not. If it's spam, the topic should be irrelevant here. I am wondering if we are creating a bias towards `Undefined` if we are marking all spam as such

Reply:
I wasn't sure either. From the spam questions in the database currently, it appears that we don't also change the topic. Maybe we should just set the topic to `None`?